### PR TITLE
Cleaning up dependencies

### PR DESCRIPTION
**Pull request by plauche**
- Archives unused projects
- Brings in `iroh-unixfs` in a trimmed down version to exclude iroh specific dependencies
- Converts all crate dependencies to workspace dependencies, tightens version specificity, narrows features when possible.

### CODE REVIEW

**Comment by plauche** [comment]
Everything under `ipfs-unixfs` was copied from https://github.com/n0-computer/beetle/tree/main/iroh-unixfs. The crate was trimmed down to exclude other iroh dependencies, any RPC calls, and references to HAMTs.

File: archive/car-utility/Cargo.lock
Comment (John-LittleBearLabs) [inline archive/car-utility/Cargo.lock:1]: I'm guessing `archive/` contains dead code.
But the "ar" in "car" is short for "archive", so... it's not 100% obvious and unambiguous what's going on here.

**Comment by ddimaria** [comment]
> failures:
> 
> ---- test_import_transmit_export_file stdout ----
> thread 'test_import_transmit_export_file' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', myceli/tests/utils/mod.rs:110:36
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> 
> ---- test_transmit_receive_dag stdout ----
> thread 'test_transmit_receive_dag' panicked at 'assertion failed: `(left == right)`
>   left: `ApplicationAPI(AvailableBlocks { cids: ["bafkreif2i55vr2qlqb4df2zj4fjqjimdmq2uolf3aa2eiqsl7xustnfhfi", "bafybeiczs7pp5etup3hpnqr2xbjc77e6xj7bedvzisqca4cp7flnxcfg2i"] })`,
>  right: `ApplicationAPI(AvailableBlocks { cids: ["bafkreiaaz47r6kzuq3qvkszenm374t5s4vbgq64ycspolugu6wrz3k2b2y", "bafkreib2tnpxikbvstnc3fpd7gihy4edhym5s6kanjmpdpqxb6c35wzb44", "bafkreieuynazhhb4jpvbjtubh424ai7e27ce3qvydvzbcnb6uxgfk3wgma", "bafkreif2i55vr2qlqb4df2zj4fjqjimdmq2uolf3aa2eiqsl7xustnfhfi", "bafkreigkcvnisvqub6yrwxbwkde2jlbqojmtsxrcptmmu272r6dsucsfyu", "bafybeiczs7pp5etup3hpnqr2xbjc77e6xj7bedvzisqca4cp7flnxcfg2i"] })`', myceli/tests/listener_test.rs:86:5
> 
> 
> failures:
>     test_import_transmit_export_file
>     test_transmit_receive_dag
> 
> test result: FAILED. 4 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.62s

**Comment by plauche** [comment]
> > failures:
> > ---- test_import_transmit_export_file stdout ----
> > thread 'test_import_transmit_export_file' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', myceli/tests/utils/mod.rs:110:36
> > note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> > ---- test_transmit_receive_dag stdout ----
> > thread 'test_transmit_receive_dag' panicked at 'assertion failed: `(left == right)`
> > left: `ApplicationAPI(AvailableBlocks { cids: ["bafkreif2i55vr2qlqb4df2zj4fjqjimdmq2uolf3aa2eiqsl7xustnfhfi", "bafybeiczs7pp5etup3hpnqr2xbjc77e6xj7bedvzisqca4cp7flnxcfg2i"] })`,
> > right: `ApplicationAPI(AvailableBlocks { cids: ["bafkreiaaz47r6kzuq3qvkszenm374t5s4vbgq64ycspolugu6wrz3k2b2y", "bafkreib2tnpxikbvstnc3fpd7gihy4edhym5s6kanjmpdpqxb6c35wzb44", "bafkreieuynazhhb4jpvbjtubh424ai7e27ce3qvydvzbcnb6uxgfk3wgma", "bafkreif2i55vr2qlqb4df2zj4fjqjimdmq2uolf3aa2eiqsl7xustnfhfi", "bafkreigkcvnisvqub6yrwxbwkde2jlbqojmtsxrcptmmu272r6dsucsfyu", "bafybeiczs7pp5etup3hpnqr2xbjc77e6xj7bedvzisqca4cp7flnxcfg2i"] })`', myceli/tests/listener_test.rs:86:5
> > failures:
> > test_import_transmit_export_file
> > test_transmit_receive_dag
> > test result: FAILED. 4 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.62s

Unfortunately I have created tests with a bit of flakiness 😭 I'll probably just rerun for now but I need to rework them for more reliability.

**Pull request merged**